src/components/timer/Timer.cpp

```diff
@@ -9,18 +9,33 @@ Timer::Timer(void* const timerData, TimerCallbackFunction_t timerCallbackFunctio
 void Timer::StartTimer(std::chrono::milliseconds duration) {
   xTimerChangePeriod(timer, pdMS_TO_TICKS(duration.count()), 0);
   xTimerStart(timer, 0);
+  expiry = xTimerGetExpiryTime(timer);
+  triggered = true;
 }
 
-std::chrono::milliseconds Timer::GetTimeRemaining() {
+// nullopt if timer stopped (StopTimer called / StartTimer not yet called)
+// otherwise TimerStatus with the ticks until/since expiry (depending on state of expired flag)
+std::optional<Timer::TimerStatus> Timer::GetTimerState() {
   if (IsRunning()) {
```

Comment:
Is it really necessary to check if the timer is running?
I would like to argue that we only check whether the timer was triggered. And if it was, we can then check whether `expiry >= xTaskGetTickCount()` or not, to calculate the distance between the two and set `expired` to either true or false.

The function could start with
```
if (!triggered) {
    return std::nullopt;
}
// ...
```

Reply:
I see, thank you :)